src/Microsoft.PowerApps.TestEngine/TestInfra/PlaywrightTestInfraFunctions.cs

```diff
@@ -307,21 +307,78 @@ public async Task HandleUserEmailScreen(string selector, string value)
         [ExcludeFromCodeCoverage]
         public async Task HandleUserPasswordScreen(string selector, string value, string desiredUrl)
         {
+            var logger = _singleTestInstanceState.GetLogger();
+
+            // Setting options fot the RunAndWaitForNavigationAsync function
             PageRunAndWaitForNavigationOptions options = new PageRunAndWaitForNavigationOptions();
+
+            // URL that should be redirected to
             options.UrlString = desiredUrl;
-            var logger = _singleTestInstanceState.GetLogger();
+
             ValidatePage();
 
             try
             {
+                // Only continue if redirected to the correct page
                 await Page.RunAndWaitForNavigationAsync(async () =>
                 {
+                    // Find the password box
                     await Page.Locator(selector).WaitForAsync();
+
+                    // Fill in the password
                     await Page.FillAsync(selector, value);
+
+                    // Submit password form
                     await this.ClickAsync("input[type=\"submit\"]");
 
-                    ValidatePage();
-                    await Page.ClickAsync("[id=\"idBtn_Back\"]");
+                    // For instances where there is a 'Stay signed in?' dialogue box
+                    try
+                    {
+                        ValidatePage();
+
+                        logger.LogDebug("Checking if asked to stay signed in.");
+
+                        // Check if we received a 'Stay signed in?' box?
+                        await Page.WaitForSelectorAsync("[id=\"KmsiCheckboxField\"]");
+                        logger.LogDebug("Was asked to 'stay signed in'.");
+
+                        // Click to stay signed in
+                        await Page.ClickAsync("[id=\"idBtn_Back\"]");
+                    }
+                    // If there is no 'Stay signed in?' box, an exception will throw; just catch and continue
+                    catch (Exception ssiException)
+                    {
+                        logger.LogDebug("Exception encountered: " + ssiException.ToString());
+
+                        // Keep record if passwordError was encountered
+                        bool hasPasswordError = false;
+
+                        try
+                        {
+                            // Check if we received a password error
+                            await Page.WaitForSelectorAsync("[id=\"passwordError\"]");
```

Comment:
Same here. If we've waited for x seconds for the SSI box, we don't need to wait extra time and we can go ahead and check the message.